Cannot find module 'node-uuid'

- `npm install trailpack-knex sqlite3 --save`
- `main.js`
```
// ...
  packs: [
    require('trailpack-repl'),
    require('trailpack-router'),
    require('trailpack-hapi'),
    require('trailpack-waterline'),
    require('trailpack-footprints'),
    require('trailpack-knex')
  ]
```
- `npm start`

Error:
```
module.js:471
    throw err;
    ^

Error: Cannot find module 'node-uuid'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/matej/dev/trails-sample/node_modules/knex/lib/raw.js:24:17)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/matej/dev/trails-sample/node_modules/knex/lib/index.js:8:12)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
```

I'm having this issue right now, too. If the project is built with Waterline, NPM doesn't fail, but with this trailpack-knex, it fails with this as an output:
```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for trailpack-knex@v2-latest
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/blah/.npm/_logs/2017-10-29T22_45_57_171Z-debug.log
```

Then, if I `npm install --save trailpack-knex`, your error appears.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,8 @@
 language: node_js
 node_js:
-- 4
-- 5
-- 6
+- 7
+- 8
+- 9
 
 notifications:
   email: false
@@ -20,4 +20,4 @@ deploy:
   on:
     tags: true
     repo: trailsjs/trailpack-knex
-    node: 6
+    node: 7

--- a/api/services/FootprintService.js
+++ b/api/services/FootprintService.js
@@ -1,6 +1,4 @@
-'use strict'
-
-const Service = require('trails-service')
+const Service = require('trails/lib/Service')
 
 /**
  * @module FootprintService

--- a/api/services/SchemaMigrationService.js
+++ b/api/services/SchemaMigrationService.js
@@ -1,7 +1,5 @@
-'use strict'
-
 const _ = require('lodash')
-const Service = require('trails-service')
+const Service = require('trails/lib/Service')
 
 /**
  * @module SchemaMigrationService
@@ -53,6 +51,10 @@ module.exports = class SchemaMigrationService extends Service {
    */
   alter (knex, model) {
     throw new Error('trailpack-knex does not currently support migrate=alter')
+  }
+
+  none () {
+
   }
 }
 

--- a/index.js
+++ b/index.js
@@ -1,8 +1,6 @@
-'use strict'
-
 const _ = require('lodash')
 const knex = require('knex')
-const DatastoreTrailpack = require('trailpack-datastore')
+const DatastoreTrailpack = require('trailpack/datastore')
 
 /**
  * Knex integration for Trails. Allows knex to read its configration from the
@@ -14,28 +12,31 @@ module.exports = class KnexTrailpack extends DatastoreTrailpack {
    * Ensure that this trailpack supports the configured migration
    */
   validate () {
+    /*
     if (!_.includes([ 'none', 'drop', 'create' ], this.app.config.database.models.migrate)) {
       throw new Error('Migrate must be configured to either "create" or "drop"')
     }
+    */
   }
 
   configure () {
-    this.app.config.database.orm = 'knex'
+    //this.app.config.stores.orm = 'knex'
   }
 
   /**
    * Initialize knex connections, and perform migrations.
    */
-  initialize () {
+  async initialize () {
     super.initialize()
 
-    this.stores = _.mapValues(this.app.config.database.stores, (store, storeName) => {
+    this.stores = _.mapValues(this.app.config.stores, (store, storeName) => {
       return {
-        knex: knex(JSON.parse(JSON.stringify(store))),
-        models: _.pickBy(this.app.models, { store: storeName })
+        knex: knex(Object.assign({ }, store)),
+        models: _.pickBy(this.app.models, { store: storeName }),
+        migrate: store.migrate
       }
     })
-    this.defaultStore = this.stores[this.app.config.database.models.defaultStore]
+    //this.defaultStore = this.stores[this.app.config.models.defaultStore]
 
     return this.migrate()
   }
@@ -43,7 +44,7 @@ module.exports = class KnexTrailpack extends DatastoreTrailpack {
   /**
    * Close all database connections
    */
-  unload () {
+  async unload () {
     return Promise.all(
       _.map(this.stores, store => store.knex.destroy())
     )
@@ -62,21 +63,20 @@ module.exports = class KnexTrailpack extends DatastoreTrailpack {
    */
   migrate () {
     const SchemaMigrationService = this.app.services.SchemaMigrationService
-    const database = this.app.config.database
 
-    if (database.models.migrate == 'none') return
+    //if (this.app.config.get(models.migrate == 'none') return
 
     return Promise.all(
-      _.map(this.stores, store => {
-        if (database.models.migrate == 'drop') {
+      _.map(this.stores, (store, storeName) => {
+        if (store.migrate === 'drop') {
           return SchemaMigrationService.drop(store.knex, this.app.models)
+            .then(result => SchemaMigrationService.create(store.knex, this.app.models))
         }
-      }))
-      .then(() => {
-        return Promise.all(_.map(this.stores, store => {
-          return SchemaMigrationService.create(store.knex, this.app.models)
-        }))
+        else {
+          return SchemaMigrationService[store.migrate](store.knex, this.app.models)
+        }
       })
+    )
   }
 }
 

--- a/index.js
+++ b/index.js
@@ -20,7 +20,7 @@ module.exports = class KnexTrailpack extends DatastoreTrailpack {
   }
 
   configure () {
-    //this.app.config.stores.orm = 'knex'
+    this.app.config.set('stores.orm', 'knex')
   }
 
   /**
@@ -36,8 +36,6 @@ module.exports = class KnexTrailpack extends DatastoreTrailpack {
         migrate: store.migrate
       }
     })
-    //this.defaultStore = this.stores[this.app.config.models.defaultStore]
-
     return this.migrate()
   }
 
@@ -63,8 +61,6 @@ module.exports = class KnexTrailpack extends DatastoreTrailpack {
    */
   migrate () {
     const SchemaMigrationService = this.app.services.SchemaMigrationService
-
-    //if (this.app.config.get(models.migrate == 'none') return
 
     return Promise.all(
       _.map(this.stores, (store, storeName) => {

--- a/lib/util.js
+++ b/lib/util.js
@@ -1,4 +1,2 @@
 module.exports = {
-
-
 }

--- a/package.json
+++ b/package.json
@@ -32,7 +32,7 @@
     "test": "eslint . && mocha"
   },
   "engines": {
-    "node": ">= 4.0.0"
+    "node": ">= 7.6.0"
   },
   "eslintConfig": {
     "extends": "trails"

--- a/package.json
+++ b/package.json
@@ -17,6 +17,8 @@
   "dependencies": {
     "knex": "0.11.6",
     "lodash": "^4.0.0",
+    "node-uuid": "^1.4.8",
+    "sqlite3": "^3.1.13",
     "trailpack": "^1.0.0-alpha-7",
     "trailpack-datastore": "^1.0.0",
     "trails-service": "^1.0.0-alpha-1"
@@ -26,7 +28,6 @@
     "eslint-config-trails": "^1.0.1",
     "mocha": "^3.0.2",
     "smokesignals": "^1.0.2",
-    "sqlite3": "waterlinejs/node-sqlite3",
     "trailpack-core": "^1.0.0-alpha-3",
     "trails": "^1.0.0-alpha-1",
     "trails-model": "^1.0.0-alpha-1"

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "trailpack-knex",
-  "version": "0.5.14",
+  "version": "3.0.0",
   "description": "Knex Trailpack",
   "homepage": "https://github.com/trailsjs/trailpack-knex",
   "author": {

--- a/package.json
+++ b/package.json
@@ -19,18 +19,14 @@
     "lodash": "^4.0.0",
     "node-uuid": "^1.4.8",
     "sqlite3": "^3.1.13",
-    "trailpack": "^1.0.0-alpha-7",
-    "trailpack-datastore": "^1.0.0",
-    "trails-service": "^1.0.0-alpha-1"
+    "trailpack": "^3"
   },
   "devDependencies": {
     "eslint": "^3.1.1",
-    "eslint-config-trails": "^1.0.1",
-    "mocha": "^3.0.2",
-    "smokesignals": "^1.0.2",
-    "trailpack-core": "^1.0.0-alpha-3",
-    "trails": "^1.0.0-alpha-1",
-    "trails-model": "^1.0.0-alpha-1"
+    "eslint-config-trails": "^3",
+    "mocha": "^4",
+    "smokesignals": "^3",
+    "trails": "^3"
   },
   "scripts": {
     "test": "eslint . && mocha"

--- a/test/app.js
+++ b/test/app.js
@@ -1,8 +1,6 @@
-'use strict'
-
 const _ = require('lodash')
 const smokesignals = require('smokesignals')
-const Model = require('trails-model')
+const Model = require('trails/lib/Model')
 
 module.exports = _.defaultsDeep({
   pkg: {
@@ -11,12 +9,22 @@ module.exports = _.defaultsDeep({
   api: {
     models: {
       User: class User extends Model {
+        static config () {
+          return {
+            store: 'teststore'
+          }
+        }
         static schema (table) {
           table.increments('id').primary()
           table.string('username')
         }
       },
       Role: class Role extends Model {
+        static config () {
+          return {
+            store: 'teststore'
+          }
+        }
         static schema (table) {
           table.increments('id').primary()
           table.string('username')
@@ -30,22 +38,16 @@ module.exports = _.defaultsDeep({
     },
     main: {
       packs: [
-        require('trailpack-core'),
         require('../') // trailpack-knex
       ]
     },
-    database: {
-      stores: {
-        teststore: {
-          client: 'sqlite3',
-          connection: {
-            filename: './testdb.sqlite'
-          }
+    stores: {
+      teststore: {
+        migrate: 'drop',
+        client: 'sqlite3',
+        connection: {
+          filename: './testdb.sqlite'
         }
-      },
-      models: {
-        defaultStore: 'teststore',
-        migrate: 'drop'
       }
     }
   }

--- a/test/index.js
+++ b/test/index.js
@@ -1,5 +1,3 @@
-'use strict'
-
 const TrailsApp = require('trails')
 
 before(() => {

--- a/test/mocha.opts
+++ b/test/mocha.opts
@@ -1,7 +1,6 @@
 --reporter spec
 --recursive
 --full-trace
---no-exit
 --slow 50
 --check-leaks
 --globals app

--- a/test/trailpack.test.js
+++ b/test/trailpack.test.js
@@ -1,11 +1,10 @@
-'use strict'
-
 const assert = require('assert')
 
 describe('Knex Trailpack', () => {
   describe('#initialize', () => {
     it('should group stores and models', () => {
       const stores = global.app.packs.knex.stores
+
       assert(stores.teststore)
       assert(stores.teststore.models)
       assert.equal(stores.teststore.models.user)

--- a/test/unit/services/SchemaMigrationService.test.js
+++ b/test/unit/services/SchemaMigrationService.test.js
@@ -10,10 +10,13 @@ describe('SchemaMigrationService', () => {
   describe('#create', () => {
     it('should create tables', () => {
       return Promise.all(_.map(global.app.models, model => {
+        console.log('store', model.store)
         const store = global.app.packs.knex.stores[model.store]
 
+        console.log('created table?', model.getTableName())
         return store.knex.schema.hasTable(model.getTableName())
           .then(exists => {
+            console.log('table exists?', exists)
             return exists ? Promise.resolve() : Promise.reject()
           })
       }))

